docs/overview/basic_concurrency.md

```diff
@@ -3,32 +3,38 @@ id: overview_basic_concurrency
+ZIO is a highly concurrent framework, powered by _fibers_, which are lightweight virtual threads that achieve massive scalability compared to threads.
 
-When you can, you should always use high-level operations, rather than working with fibers directly. For the sake of completeness, this section introduces both fibers and some of the high-level operations built on them.
+ZIO augments the modern virtual threading model with pervasive, automatic, resource-safe interruption, which allows ZIO to instantly and deeply cancel running computations when their results are no longer needed, making ZIO applications automatically globally-efficient.
+
+In this section, you will learn the basics of fibers, and become acquainted with some of the powerful high-level operators that are powered by fibers.
 
 ## Fibers
 
-ZIO's concurrency is built on _fibers_, which are lightweight "green threads" implemented by the ZIO runtime system.
+All effects in ZIO are executed by _some_ fiber. If you did not create the fiber, then the fiber was created by some operation you are using (if the operation is concurrent or parallel), or by the ZIO runtime system.
 
-Unlike operating system threads, fibers consume almost no memory, have growable and shrinkable stacks, don't waste resources blocking, and will be garbage collected automatically if they are suspended and unreachable.
+Even if you only write "single-threaded" code, with no parallel or concurrent operations, then there will be at least one fiber: the "main" fiber that executes your effect.
 
-Fibers are scheduled by the ZIO runtime and will cooperatively yield to each other, which enables multitasking, even when operating in a single-threaded environment (like JavaScript, or even the JVM when configured with one thread).
+Like operating system-level threads, ZIO fibers have a well-defined lifecycle, defined by the effect they are executing.
 
-All effects in ZIO are executed by _some_ fiber. If you did not create the fiber, then the fiber was created by some operation you are using (if the operation is concurrent or parallel), or by the ZIO runtime system.
+Every fiber exits with failure or success, depending on whether the effect it is executing fails or succeeds.
 
-Even if you only write "single-threaded" code, with no parallel or concurrent operations, then there will be at least one fiber: the "main" fiber that executes your effect.
+Also like operating system threads, ZIO fibers have unique identities, stacks (including stack traces), local state, and a status (such as _done_, _running_, or _suspended_).
+
+Unlike operating system threads, ZIO fibers consume almost no memory, they have dynamic stacks that grow and shrink, they don't waste operating system threads with blocking operations, they can be safely interrupted at any point in time, they are strongly typed, they have enumerable child fibers, and they will be garbage collected automatically if they are suspended and cannot be reactivated.
```

Comment:
Maybe it makes sense to make this into a list. Might be a bit hard to read.

Reply:
Fixed